app/src/main/java/de/westnordost/streetcomplete/quests/lanes/AddLanes.kt

```diff
@@ -0,0 +1,72 @@
+package de.westnordost.streetcomplete.quests.lanes
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmFilterQuestType
+
+class AddLanes : OsmFilterQuestType<LanesAnswer>() {
+
+    override val elementFilter = """
+        ways with
+          highway ~ ${ROADS_WITH_LANES.joinToString("|")}
+          and (!lanes or lanes !~ [1-9][0-9]*)
+    """
+    override val commitMessage = "Add road lanes"
+    override val wikiLink = "Key:lanes"
+    override val icon = R.drawable.ic_quest_street_lanes
+    override val isSplitWayEnabled = true
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_lanes_title
+
+    override fun createForm() = AddLanesForm()
+
+    override fun applyAnswerTo(answer: LanesAnswer, changes: StringMapChangesBuilder) {
+
+        changes.addOrModify("lanes", answer.total.toString())
+
+        val isUnmarked = answer !is UnmarkedLanes
+        // if there is just one lane, the information whether it is marked or not is irrelevant
+        // (if there are no more than one lane, there are no markings to separate them)
+        if (answer.total == 1) changes.deleteIfExists("lane_markings")
+        else if (isUnmarked)   changes.modifyIfExists("lane_markings", "yes")
+        else                   changes.addOrModify("lane_markings", "no")
+
+        val hasCenterLeftTurnLane = answer is MarkedLanesSides && answer.centerLeftTurnLane
+        if (hasCenterLeftTurnLane) {
+            changes.addOrModify("lanes:both_ways", "1")
+            changes.addOrModify("turn:lanes:both_ways","left")
+        } else {
+            changes.deleteIfExists("lanes:both_ways")
+            changes.deleteIfExists("turn:lanes:both_ways")
+        }
+
+        when(answer) {
+            is MarkedLanes -> {
+                if (answer.count == 1) {
+                    changes.deleteIfExists("lanes:forward")
+                    changes.deleteIfExists("lanes:backward")
+                } else {
+                    changes.modifyIfExists("lanes:forward", (answer.count / 2).toString())
```

Comment:
If I am reading correctly it is displayed as `answer.count - answer.count / 2`

From `setTotalLanesCount` 

```
            leftSide = lanes / 2
            rightSide = lanes - lanes / 2
```

(this time I started from looking at source code, it may have noticeable effects even for odd lane count, but seemed suspicious)

Reply:
`answering marked lanes with previous marked for each side answer` test seems to ensure that it is not a problem.

Reply:
In general the `answer.count` should always be an even number, except for oneway roads.
Lanes count passe to `setTotalLanesCount` may be odd, because when selecting unmarked lanes, it is allowed to specify an odd number.